src/pomodoro/useTimerReducer.ts

```diff
@@ -0,0 +1,88 @@
+import { useReducer } from 'react';
+import { TimerSchema } from '../common/timerSchemaTypes';
+import { DEFAULT_TIMER_SCHEMA } from '../common/constants';
+
+const MINUTE_IN_SECONDS = 60;
+const FIVE_MINUTES_IN_SECONDS = MINUTE_IN_SECONDS * 5;
+const TWENTY_MINUTES_IN_SECONDS = MINUTE_IN_SECONDS * 20;
+
+const INITIAL_OBJECT: State = {
+  time: DEFAULT_TIMER_SCHEMA.timerDuration * MINUTE_IN_SECONDS,
+  isActivated: false,
+  timerSchema: DEFAULT_TIMER_SCHEMA,
+};
+
+enum ActionTypes {
+  Start = 'start',
+  Stop = 'stop',
+  Reset = 'reset',
+  DecreaseTime = 'decreaseTime',
+  IncreaseTimeByOneMin = 'increaseTimerByOneMin',
+  IncreaseTimeByFiveMin = 'increaseTimerByFiveMin',
+  IncreaseTimeByTwentyMin = 'increaseTimerByTwentyMin',
+  ChangeTimerSchema = 'changeTimerSchema',
+}
+
+type State = {
+  time: number;
+  isActivated: boolean;
+  timerSchema: TimerSchema;
+};
+
+type Action = {
+  type: ActionTypes;
+};
+
+function timerReducer(state: State, action: Action): State {
+  switch (action.type) {
+    case ActionTypes.Start:
+      return { ...state, isActivated: true };
+    case ActionTypes.Stop:
+      return { ...state, isActivated: false };
+    case ActionTypes.Reset:
+      return { ...state };
+    case ActionTypes.DecreaseTime:
+      return { ...state, time: state.time - 1 };
+    case ActionTypes.IncreaseTimeByOneMin:
+      return { ...state, time: state.time + MINUTE_IN_SECONDS };
+    case ActionTypes.IncreaseTimeByFiveMin:
+      return { ...state, time: state.time + FIVE_MINUTES_IN_SECONDS };
+    case ActionTypes.IncreaseTimeByTwentyMin:
+      return { ...state, time: state.time + TWENTY_MINUTES_IN_SECONDS };
```

Comment:
This approach would require you to keep adding cases if you want to increment time by a different amount. Change it to one case that takes a payload.